src/neuroglancer/annotation/bounding_box.ts

```diff
@@ -410,4 +415,54 @@ registerAnnotationTypeRenderHandler(AnnotationType.AXIS_ALIGNED_BOUNDING_BOX, {
       // vec3.transformMat4(position, annotation.point, objectToData);
     }
   },
+  getRepresentativePoint: (objectToData, ann, partIndex) => {
+    let repPoint= vec3.create();
+    // if the full object is selected pick the first corner as representative
+    if (partIndex==FULL_OBJECT_PICK_OFFSET){
+      vec3.transformMat4(repPoint, ann.pointA, objectToData);
+    }
+    else if (partIndex >= CORNERS_PICK_OFFSET && partIndex < EDGES_PICK_OFFSET) {
+      // picked a corner
+      // FIXME: figure out how to return corner point
+      vec3.transformMat4(repPoint, ann.pointA, objectToData);
+    } else if (partIndex >= EDGES_PICK_OFFSET && partIndex < FACES_PICK_OFFSET) {
+      // FIXME: can't figure out how to resize based upon edge grabbed
+      vec3.transformMat4(repPoint, ann.pointA, objectToData);
+      // snapPositionToCorner(repPoint, objectToData, corners, 5);
+    }
+    else{ // for now faces will move the whole object so pick the first corner
+      vec3.transformMat4(repPoint, ann.pointA, objectToData);
+    }
+    return repPoint
+  }, 
+  updateViaRepresentativePoint: (oldAnnotation: AxisAlignedBoundingBox, position: vec3, dataToObject: mat4, partIndex: number) => {
+    let newPt = vec3.transformMat4(vec3.create(), position, dataToObject);
```

Comment:
You can instead use the object spread notation let baseBox = <AxisAlignedBoundingBox>{...oldAnnotation};

Same thing applies to other annotation types.

Reply:
In addition to being shorter, this also ensures that we copy all of the fields.  Is there a reason why you are setting the id to the empty string?

Reply:
I thought you had said that the annotations should be atomic, so I thought the new annotation should get a new ID, and I thought that ID got assigned when i added the new annotation to the layer.  Is my thinking off?

Reply:
The id should stay the same.  One of the purposes of the id is to allow an annotation to be uniquely identified even as it is modified.